src/closure.rs

```diff
@@ -23,31 +23,31 @@ impl Closure {
             _invocation_hint: *mut c_void, marshal_data: *mut c_void)
         {
             let values = slice::from_raw_parts(param_values as *const _, n_param_values as usize);
-            let callback: &Box<F> = mem::transmute(marshal_data as *mut _);
+            let callback: Box<F> = Box::from_raw(marshal_data as *mut _);
```

Comment:
If you want to omit the mem::forget() below, you could also do `let callback: &F = mem::transmute(marshal_data)`